portal/v2/src/ClusterDetailListComponents/Overview.tsx

```diff
@@ -0,0 +1,154 @@
+    installStatus: 'Installation Status'
+}
+
+function ClusterDetailCell(
+    value: any,
+    ): any {
+        if (typeof (value.value) == typeof (" ")) {
+            return <Stack.Item styles={value.style}>
+            <Text styles={value.style} variant={'medium'}>{value.value}</Text>
+            </Stack.Item>
+        }
+    }
+    
+export class OverviewComponent extends Component<OverviewComponentProps, IOverviewComponentState> {
+    
+    constructor(props: OverviewComponentProps | Readonly<OverviewComponentProps>) {
+        super(props);
+    }
+    
+    public render() {
+        const headerEntries = Object.entries(clusterDetailHeadings)
+        const filteredHeaders: Array<[string, any]> = []
+        if (this.props.item.length != 0) {
+            headerEntries.filter((element: [string, any]) => {
+                if (this.props.item[element[0]] != null &&
+                    this.props.item[element[0]].toString().length > 0) {
+                        filteredHeaders.push(element)
+                    }
+            })
+            return (
+                <Stack styles={contentStackStylesNormal}>
+                    <Text variant="xxLarge">{this.props.clusterName}</Text>
+                    <Stack horizontal>
+                        <Stack styles={KeyColumnStyle}>
+                        {filteredHeaders.map((value: [string, any], index: number) => (
+                            <ClusterDetailCell style={KeyStyle} key={index} value={value[1]} />
+                            )
+                        )}
+                        </Stack>
+                        
+                        <Stack styles={KeyColumnStyle}>
+                        {Array(filteredHeaders.length).fill(':').map((value: [string], index: number) => (
+                            <ClusterDetailCell style={KeyStyle} key={index} value={value} />
+                            )
+                        )}
+                        </Stack>
+                        
+                        <Stack styles={ValueColumnStyle}>
+                        {filteredHeaders.map((value: [string, any], index: number) => (
+                            <ClusterDetailCell style={ValueStyle}
+                            key={index}
+                            value={this.props.item[value[0]]} />
+                            )
+                        )}
+                        </Stack>
+                    </Stack>
+                </Stack>
+            );
+        } else {
+            return (
+                <Stack>
+                <Shimmer styles={headShimmerStyle} shimmerElements={headerShimmer} width="25%"></Shimmer>
+                {headerEntries.map(header => (
+                    <Shimmer key={header[0]} styles={ShimmerStyle} shimmerElements={rowShimmer} width="75%"></Shimmer>
+                    )
+                )}
+                </Stack>
+                )
+            }
+        }
+    }
+                                
```

Comment:
nit: no new line, I believe there is a space on this line
```suggestion

```

Reply:
@s-amann Suggestion was even better as it applied DRY, but if it can't be applied, no more debate :)

Regarding a single long function, IMHO it is always cleaner/better to have a short function that gives you an overall idea about what the code does without going further into the details. I think good encapsulation/high cohesion starts with having short functions. 

The clear advantage I see with the multiple functions I suggested is that you can understand what "MachinesFromMachineList" does reading just 18 LOC (the function itself) and later you can go read each of the narrow functions to understand the details. I did "comment PR" instead of "suggest changes" as there is nothing wrong with the code and this is just a personal suggestion to improve it a bit (from my personal point of view). There is an interesting talk about [Clean Code functions](https://www.youtube.com/watch?v=QvrNwfuWbCQ) from Uncle Bob.

So in the end is up to you 🙂

Reply:
Done